app/scripts/views/mainTemplate.js

```diff
@@ -62,30 +62,38 @@
           this.completeSamplesList = _.keys(iViz.getCasesMap('sample')).sort();
         }
       },
-      /*'patientmap':function(val){
-        this.completePatientsList =  _.keys(val);
-      },
-      'samplemap' : function(val){
-        this.completeSamplesList =  _.keys(val);
-      },*/
-      'messages': function(val) {
-        _.each(this.groups, function(group) {
-          dc.renderAll(group.id);
-        });
-        this.updateGrid();
+      'messages': function(ChartsIds) {
+        if(ChartsIds.length>0){
+          _.each(this.groups, function(group) {
+            dc.renderAll(group.id);
+          });
+          this.updateGrid(ChartsIds);
+          this.messages.length=0;
+        }
       }
     }, methods: {
-      updateGrid: function() {
+      updateGrid: function(ChartsIds) {
+        var self_ = this;
+        function SortByNumber(a, b){
```

Comment:
Can we make this a singleton? Maybe put it into util or maybe just inside the template vue methods object?